apps/emqx_management/src/emqx_mgmt_api_publish.erl

```diff
@@ -272,6 +272,8 @@ is_ok_deliver({_NodeOrShare, _MatchedTopic, {error, _}}) -> false.
 %% No preceding payload format indicator to compare against.
 %% Content-Type check should be done at HTTP layer but not here.
 %% 153                Payload format invalid                  400
+publish_result_to_http_reply(#message{topic = <<"$delayed/", _/binary>>} = Message, []) ->
```

Comment:
Thinking a bit more about this, more future-proof I guess would be for the `emqx_delayed` hook to signal to the broker that "look, this message is taken care of, answer with `RC_SUCCESS` to the client", and the result of `publish/1` would reflect that.

Reply:
Makes sense.  However, at this point, the only thing that comes back from the delayed message hook is an empty destination list, so currently we can't return, for example, the message with special headers signaling the reason code to the hook without changing this hook's API.